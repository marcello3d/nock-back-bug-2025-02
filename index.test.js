// Copyright 2025 Descript, Inc
const { back } = require("nock");
const path = require("path");

describe("nock.back", () => {
  back.fixtures = path.join(__dirname, "__nocks__");

  it("should work", async () => {
    const { nockDone } = await back(`test1.json`);
    const response = await fetch("https://api.descript.com");
    await expect(response.text()).resolves.toMatchSnapshot();
    nockDone();
  });

  it("handles parallel requests", async () => {
    const { nockDone } = await back(`test2.json`);
    const responses = await Promise.all([
      fetch("https://api.descript.com"),
      fetch("https://api.descript.com"),
    ]);
    expect(responses.length).toBe(4);
    nockDone();
  });
});
