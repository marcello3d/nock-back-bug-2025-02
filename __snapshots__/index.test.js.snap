// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nock.back should work 1`] = `"ok"`;
